v4/newrelic/segments.go

```diff
@@ -214,18 +216,58 @@ func (s *DatastoreSegment) End() {
 	if s.StartTime.isEnded() {
 		return
 	}
+
+	s.addAttributes(s.StartTime.Span.SetAttribute)
 	s.StartTime.Span.SetName(s.name())
 	s.StartTime.end()
 }
 
+func (s *DatastoreSegment) addAttributes(set func(string, interface{})) {
+	set("db.system", valOrUnknown(string(s.Product)))
+	set("db.statement", valOrUnknown(s.statement()))
+	set("db.operation", valOrUnknown(s.Operation))
+
+	if net.ParseIP(s.Host) != nil {
+		set("net.peer.ip", s.Host)
+	} else {
+		set("net.peer.name", valOrUnknown(s.Host))
+	}
+	if s.PortPathOrID != "" {
+		if port, err := strconv.Atoi(s.PortPathOrID); err == nil {
+			set("net.peer.port", port)
+		}
+	}
+
+	switch s.Product {
+	case DatastoreCassandra:
+		set("db.cassandra.keyspace", valOrUnknown(s.DatabaseName))
+	case DatastoreRedis:
+		set("db.redis.database_index", valOrUnknown(s.DatabaseName))
+	case DatastoreMongoDB:
+		set("db.mongodb.collection", valOrUnknown(s.DatabaseName))
+	default:
+		set("db.name", valOrUnknown(s.DatabaseName))
+	}
+}
+
 func (s *DatastoreSegment) name() string {
+	return s.statement()
+}
+
+func valOrUnknown(v string) string {
+	if v == "" {
+		return "unknown"
+	}
+	return v
+}
+
+func (s *DatastoreSegment) statement() string {
 	pq := s.ParameterizedQuery
 	if pq == "" {
-		coll := s.Collection
-		if "" == coll {
-			coll = "unknown"
-		}
-		pq = "'" + s.Operation + "' on '" + coll + "' using '" + string(s.Product) + "'"
+		op := valOrUnknown(s.Operation)
+		coll := valOrUnknown(s.Collection)
+		prod := valOrUnknown(string(s.Product))
+		pq = "'" + op + "' on '" + coll + "' using '" + prod + "'"
```

Comment:
This change updated the names of several spans in our tests. I feel like using `'unknown'` instead of `''` in places is easier to read and understand.